modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/statistic/ItStatisticTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.statistic;
+
+import static org.apache.ignite.internal.sql.engine.util.QueryChecker.containsRowCount;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.ignite.internal.sql.BaseSqlIntegrationTest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/** Integration test to check SQL statistics. */
+public class ItStatisticTest extends BaseSqlIntegrationTest {
+    private SqlStatisticManagerImpl sqlStatisticManager;
+
+    @BeforeAll
+    void beforeAll() {
+        sqlStatisticManager = (SqlStatisticManagerImpl) queryProcessor().sqlStatisticManager();
+        sql("CREATE TABLE t(ID INTEGER PRIMARY KEY, VAL INTEGER)");
+    }
+
+    @AfterAll
+    void afterAll() {
+        sql("DROP TABLE IF EXISTS t");
+    }
+
+    @Test
+    public void testStatisticsRowCount() throws InterruptedException {
```

Comment:
there is no synchronization between updating table size in background and execution of next statement which expects the size to be updated. This means this test is in fact flaky. Try to add delay just before `tableSizeMap.computeIfPresent(tableId, (k, v) -> new ActualSize(size, currTs));` to see how it will behave

Reply:
added